### PR TITLE
[ready for core team to review] CRM-20614 Don't download export file until asked

### PR DESCRIPTION
**Pull request by pradpnayak**
----------------------------------------
* CRM-20614: Donot Download Export file for Financial Batch
  https://issues.civicrm.org/jira/browse/CRM-20614

### CODE REVIEW

File: CRM/Batch/BAO/Batch.php [hunk 1]
```diff
@@ -594,6 +596,7 @@ public static function exportFinancialBatch($batchIds, $exportFormat) {
     else {
       CRM_Core_Error::fatal("Could not locate exporter: $exporterClass");
     }
+    $exporter->_isDownloadFile = $downloadFile;
```
Comment (colemanw) [inline CRM/Batch/BAO/Batch.php]: I don't see the property `$_isDownloadFile` declared in that class.

**Review by colemanw — CHANGES_REQUESTED** [review]
Overall this looks safe, but recommend class variables be explicitly defined.

File: CRM/Batch/BAO/Batch.php (hunk 1 above)
Comment (pradpnayak) [inline CRM/Batch/BAO/Batch.php]: @colemanw I have pushed the changes. Can you please review?

Comment (colemanw) [inline CRM/Batch/BAO/Batch.php]: Have you tried it? I think this will give a fatal error trying to set a protected property.

Comment (pradpnayak) [inline CRM/Batch/BAO/Batch.php]: oops sorry, Extension had a over-ridden copy of this file therefore i didn't get the error while testing. I have updated the changes now.

**Comment by pradpnayak** [comment]
@civicrm-builder test this please

**Comment by monishdeb** [comment]
@colemanw I have tested the patch and it works fine now. Can you please check now?

**Comment by pradpnayak** [comment]
@colemanw  i have rebased the PR. Can you please review?

File: CRM/Batch/BAO/Batch.php [hunk 2]
```diff
@@ -576,8 +576,10 @@ public static function displayTotals($actual, $expected) {
    *   Associated array of batch ids.
    * @param string $exportFormat
    *   Export format.
+   * @param bool $downloadFile
+   *   Download export file?.
    */
-  public static function exportFinancialBatch($batchIds, $exportFormat) {
+  public static function exportFinancialBatch($batchIds, $exportFormat, $downloadFile) {
```
Comment (mlutfy) [inline CRM/Batch/BAO/Batch.php]: Add a default value to `downloadFile` to keep compatibility with older versions?

Comment (monishdeb) [reply CRM/Batch/BAO/Batch.php]: @mlutfy ```exportFinancialBatch(...) is used only in one place as I grep`ed in codebase
```
Monishs-MacBook-Pro:civicrm monish$ grep -irn "exportFinancialBatch" CRM api/
CRM/Batch/BAO/Batch.php:45:   * Not sure this is the best way to do this. Depends on how exportFinancialBatch() below gets called.
CRM/Batch/BAO/Batch.php:600:  public static function exportFinancialBatch($batchIds, $exportFormat, $downloadFile) {
CRM/Financial/Form/Export.php:178:    CRM_Batch_BAO_Batch::exportFinancialBatch($batchIds, $this->_exportFormat, $this->_downloadFile);
```

So I think its not required to provide a default value. What do you say?

**Comment by monishdeb** [comment]
@colemanw @mlutfy is there anything left to fix, to get this PR merged?

**Comment by colemanw** [comment]
* ([`r-jira`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-jira)) __Pass__
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __Pass__
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __Pass__
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __Pass__
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __Pass__
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))__Pass__ - tested by Monish
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __Pass__
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __Pass__

**Review by colemanw — APPROVED** [review]

**Pull request merged**